src/YarpUtilities/src/RosPublisher.cpp

```diff
@@ -0,0 +1,482 @@
+{
+    std::string_view printPrefix = "[RosPublisher::publishJointStates] ";
+    
+    if (!m_pimpl->initialized)
+    {
+        std::cerr << printPrefix << "Please call initialize before publishing." << std::endl;
+        return false;
+    }
+        
+    auto& pub = m_pimpl->jointStatePublisher;
+    if (pub.ptr == nullptr)
+    {
+        std::cerr << printPrefix << "Joint state publisher was not configured. Unable to publish joint states." << std::endl;
+        return false;
+    }
+        
+    if ( (jointList.size() != jointPositions.size()) || 
+         (jointList.size() != jointVelocities.size()) ||
+         (jointList.size() != jointEfforts.size()))
+    {
+        std::cerr << printPrefix << "Joint states size mismatch. Unable to publish joint states." << std::endl;
+        return false;
+    }
+        
+    pub.msg.header.seq++;
+    pub.msg.header.stamp = m_pimpl->getTimeStampFromYarp();
+    
+    pub.msg.name.clear();
+    pub.msg.position.clear();
+    pub.msg.velocity.clear();
+    pub.msg.effort.clear();
+    
+    for (std::size_t idx = 0; idx < jointList.size(); idx++)
+    {
+        pub.msg.name.emplace_back(jointList(idx));
+        pub.msg.position.emplace_back(jointPositions(idx));
+        pub.msg.velocity.emplace_back(jointVelocities(idx));
+        pub.msg.effort.emplace_back(jointEfforts(idx));
+    }
+    
+    pub.ptr->write(pub.msg);
+    return true;
+}
+
+bool RosPublisher::publishWrench(const std::string& frame, 
+                                 BipedalLocomotion::GenericContainer::Vector<const double>::Ref wrench6d)
+{
+    std::string_view printPrefix = "[RosPublisher::publishWrench] ";
+    
+    if (!m_pimpl->initialized)
+    {
+        std::cerr << printPrefix << "Please call initialize before publishing." << std::endl;
+        return false;
+    }
+    
+    if (m_pimpl->wrenchPublisherMap.find(frame) == m_pimpl->wrenchPublisherMap.end())
+    {
+        std::cerr << printPrefix << "Frame does not exist. Please add wrench publisher before publishing." << std::endl;
+        return false;
+    }
+            
+    auto& pub = m_pimpl->wrenchPublisherMap.at(frame);    
+    if (pub.ptr == nullptr)
+    {
+        std::cerr << printPrefix << "Wrench publisher was not configured. Unable to publish wrench stamped messages." << std::endl;
+        return false;
+    }
+        
+    if (wrench6d.size() != 6)
+    {
+        std::cerr << printPrefix << "Expecting a 6d input vector. Unable to publish wrench." << std::endl;
+        return false;
+    }
+    
+    pub.msg.header.seq++;
+    pub.msg.header.frame_id = frame;
+    pub.msg.header.stamp = m_pimpl->getTimeStampFromYarp();
+    pub.msg.wrench.force.x = wrench6d(0);
+    pub.msg.wrench.force.y = wrench6d(1);
+    pub.msg.wrench.force.z = wrench6d(2);
+    
+    pub.msg.wrench.torque.x = wrench6d(3);
+    pub.msg.wrench.torque.y = wrench6d(4);
+    pub.msg.wrench.torque.z = wrench6d(5);
```

Comment:
Maybe it is worth writing somewhere this expected serialization.

Reply:
Addressed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/156/commits/d57a286f4780f87c335ff0b3b91ca01443ffa069